### ISSUE TITLE
Throw exception on TemplateCacheWarmer failure

### ISSUE THREAD

**Issue by umulmrum** [symfony/symfony#41922]
**Description**
The TemplateCacheWarmer has been ignoring errors since its first release in 2015. But I wonder what the rationale behind this decision is, as a) I'd like errors to pop up as early as possible (i.e. in a CI pipeline where the cache is warmed), and b) other cache warmers do throw exceptions, so ignoring errors doesn't seem to be a general rule in warmers.

Currently all templates with errors are skipped, so errors only appear during runtime (this might also trigger errors when deploying into a readonly filesystem but I didn't test it).

Unfortunately the comment in the empty catch block doesn't tell us why errors are ignored.

tl;dr: I ask to consider the TemplateCacheWarmer to fail on error.

**Comment by stof** [comment]
Well, if you want to see template syntax errors in CI, run the `lint:twig` command in your CI.

Also, for templates, making it fail when a template cannot be parsed without errors would require adding a feature to exclude templates from the cache warmer. For instance, the templates shipped in DoctrineBundle for the profiler panels are relying on Twig functions provided by WebProfilerBundle. In a prod environment, you will not have WebProfilerBundle and so not those functions. This would make TemplateCacheWarmer fail when warming up the DoctrineBundle templates (that will never be used in prod as they are used only when you render the profiler, but they are still in the bundle).

**Comment by umulmrum** [comment]
Ah, I wasn't aware of your second argument, thanks for the pointer. But that also means that `lint:twig` can only be run in `dev` and `test` environments (where the WebProfilerBundle is active by default), and it might happen that arbitrary templates also depend on dev-only functions and explode in `prod`, right? Solutions to this might be more complex for developers so I'm not sure what the "perfect" way could look like. At the very least the comment in the catch block could be improved, I'll submit a PR for this later.
